frontend/src/pages/upload/UploadPage.tsx

```diff
@@ -10,6 +14,55 @@ import {
 import { ExperimentalMeasurementDto } from "../../api/types";
 import "../../components/Spinner.scss";
 
+type CompoundMap = { [compoundId: string]: ExperimentalMeasurementDto[] };
+
+function groupCompounds(mes: ExperimentalMeasurementDto[]): CompoundMap {
+  return mes.reduce((rv: CompoundMap, x) => {
+    // eslint-disable-next-line no-param-reassign
+    (rv[x.compoundId] = rv[x.compoundId] || []).push(x);
+    return rv;
+  }, {});
+}
+
+function hashCode(str: string) {
+  let hash = 0;
+  for (let i = 0; i < str.length; i++) {
+    const chr = str.charCodeAt(i);
+    // eslint-disable-next-line no-bitwise
+    hash = ((hash << 5) - hash) + chr;
+    // eslint-disable-next-line no-bitwise
+    hash |= 0; // Convert to 32bit integer
```

Comment:
It actually looks like bitwise `or` with 0 which should do nothing. What is going on here?

Reply:
:putya: don't know either, just copy-pasted it from the stackoverflow :putya:
will figure out

Reply:
https://stackoverflow.com/questions/6798111/bitwise-operations-on-32-bit-unsigned-ints
just some JavaScript nonsense, nvm. 

this code has been been written ~copy-pasted~ once and will be touched never again